apps/aefate/src/aefa_fate_op.erl

```diff
@@ -1239,6 +1248,89 @@ spend(Arg0, Arg1, ES0) ->
             aefa_fate:abort({value_does_not_match_type, Other, address}, ES2)
     end.
 
+create(Arg0, Arg1, Arg2, Arg3, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {[?FATE_FATE_CODE(Code), InitArgsTypes, Value], ES1} =
+        get_op_args([Arg1, Arg2, Arg3], ES0),
+
+    {ContractPK, ES2} = create_contract(Code, Value, ES1),
+    ReplaceInitResult
+        = fun(ES0_) ->
+                  %% init returns unit, so we get rid of it
+                  {{tuple, {}}, ES1_} = get_op_arg({stack, 0}, ES0_),
+                  write(Arg0, ?FATE_CONTRACT(ContractPK), ES1_)
+          end,
+    InitName = aeb_fate_code:symbol_identifier(<<"init">>),
+    case remote_call_common(
+           ?FATE_CONTRACT(ContractPK),
+           InitName,
+           InitArgsTypes,
+           ?FATE_TYPEREP({tuple, []}),
+           0,
+           no_gas_cap,
+           unprotected,
+           ReplaceInitResult,
+           ES2)
+    of
+        {ok, ES3} ->
+            {jump, 0, ES3};
+        {failed_protected_call, ES3} ->
+            {next, write(Arg0, make_none(), ES3)}
+    end.
+
+create_contract(Code, Amount, ES0) ->
+    Current = aefa_engine_state:current_contract(ES0),
+
+    AS0              = aefa_engine_state:chain_api(ES0),
+    {ok, Nonce, AS1} = aefa_chain_api:next_nonce(Current, AS0),
+
+    Contract0  = aect_contracts:new(
+                     Current, Nonce,
+                     #{vm => aefa_engine_state:vm_version(ES0)
```

Comment:
Make sure to allow only FATE2

Reply:
This is a helper function, the CREATE handler makes this check

Reply:
OK!